trace/context.go

```diff
@@ -25,19 +25,30 @@ func ContextWithSpan(parent context.Context, span Span) context.Context {
 	return context.WithValue(parent, currentSpanKey, span)
 }
 
+// ContextWithSpanContext returns a copy of parent with sc as the current
+// Span. The Span implementation that wraps sc is non-recording and performs
+// no operations other than to return sc as the SpanContext from the
+// SpanContext method.
+func ContextWithSpanContext(parent context.Context, sc SpanContext) context.Context {
```

Comment:
Do we intend this to be public on our API or could this be limited in any way to tests only?

The same applies for `ContextWithRemoteSpanContext`.

Reply:
`ContextWithRemoteSpanContext` is a continuation of the pre-existing function prior to changing how [the parent span is stored in a context](https://github.com/open-telemetry/opentelemetry-go/pull/1731) and it, or `ContextWithSpanContext`, will be need by [any propagator](https://github.com/open-telemetry/opentelemetry-go-contrib/search?q=ContextWithRemoteSpanContext).

Based on this, this, `ContextWithRemoteSpanContext`, or a new way of [wrapping a `SpanContext` in a `Span`](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.1.0/specification/trace/api.md#wrapping-a-spancontext-in-a-span) is definitely going to be needed. It seemed like a good idea to preserve `ContextWithRemoteSpanContext` as it will cause the least amount of frustration, but what I found here is there are going to be situations where we want to wrap the `SpanContext` and not have it be marked as a remote span. I can see users that want to propagate external requests different than internal requests wanting this functionality as well. Based on this it seems like exporting both makes sense to me.